src/btree/bt_curprev.c

```diff
@@ -262,6 +277,19 @@ __cursor_fix_prev(WT_CURSOR_BTREE *cbt, bool newpage, bool restart)
     if (F_ISSET(&cbt->iface, WT_CURSTD_KEY_ONLY))
         return (0);
 
+    if (F_ISSET(&cbt->iface, WT_CURSTD_BOUND_LOWER)) {
+        WT_ASSERT(session, WT_DATA_IN_ITEM(&cbt->iface.lower_bound));
```

Comment:
Can we de-duplicate this code by moving it to a function? I see we've effectively got the same code in two parts of the file, similarly I think we can merge it with the cur_next function by providing a direction.